### PR TITLE
implement driver.Connector

### PR DESCRIPTION
**Pull request by james-lawrence**
first pass to get initial review, and open the discussion on how the connector should be within pgx.

related to https://github.com/jackc/pgx/issues/380

### CODE REVIEW

File: stdlib/stdlibutil_test.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+package stdlib_test
+
+import (
+	"database/sql"
+	"testing"
+)
+
+// this file contains utility functions for tests that differ between versions.
+func openDB(t *testing.T) *sql.DB {
```
Comment (james-lawrence) [inline stdlib/stdlibutil_test.go]: TODO: add a test for sql.Open explicitly.

**Comment by jackc** [comment]
Overall structure seems reasonable to me.

Only thing that threw me for a loop was the `OptionOpenDB` pattern. I've rarely seen that style used before and we are not using it anywhere else in pgx. I suppose it is a slightly superior way to allow future extension than a configuration struct like we do elsewhere. It bothers me a little to add another style, but I guess it's worth it.

LGTM so far.

**Comment by james-lawrence** [comment]
@jackc i'll probably handle my TODO next weekend is there anything else you'd want done in order to merge?

**Comment by jackc** [comment]
Looks good to me.

**Comment by johanbrandhorst** [comment]
Bump?

**Comment by jackc** [comment]
Sorry. For some reason I didn't realize this PR was ready. Merged.

**Comment by james-lawrence** [comment]
all good was my bad kind of forgot about it myself =)

**Pull request merged**